articles/getting-started-using-admob-with-react-native-expo/index.md

```diff
@@ -0,0 +1,487 @@
+
+### STEP 2: Setup Admob on Expo
+To set up Admob on Expo, we will need to export the previous project from the Expo snack. This is due to the fact that we are not authorized to access the `app.json` to use in environment variables. Therefore, we need to [export the entire expo project](https://snack.expo.io/2y6z09xog) as directed in the screenshot below:
+
+![export expo project from snack](/engineering-education/getting-started-using-admob-with-react-native-expo/11-export-project-from-snack.png)
+
+After exporting to the local system, we need to install all the required packages as mentioned in the requirement section above. First, we are going to install Expo CLI. For that, we need to open the terminal in the project directory and run the following command:
+
+```bash
+yarn ; yarn global add expo-cli 
+```
+
+Now that our expo-cli is installed, we need to start setting up Admod. For that, we need to open the `app.json` file and add the **Admob App ID** to the `config` section as shown in the code snippet below:
+
+```json
+    "android": {
+      "adaptiveIcon": {
+        "foregroundImage": "./assets/adaptive-icon.png",
+        "backgroundColor": "#FFFFFF"
+      },
+      "config": {
+        "googleMobileAdsAppId":"ca-app-pub-2547344479047582~xxxxxxxxx"
+      }
+     
+    },
+```
+
+Now, it is time to install the expo Admob package i.e. **expo-ads-admob**. For that, we need to run the following command in our project terminal: 
+
+```json
+expo install expo-ads-admob
+```
+
+After everything is installed, we can go ahead and run the app using the following command in the project terminal:
+
+```jsx
+expo start
+```
+
+After this, we need to choose the required emulator, that is, Android to run the project on. Then on a successful build, we will get the app running in our emulator as shown in the screenshot below:
+
+![running react native expo  in local](/engineering-education/getting-started-using-admob-with-react-native-expo/12-running-react-native-expo-in-local.png)
+
+### STEP 3: Using Admob Banner
+Now, we are going to use the Banner Ad that we created before in Admob Account in our Home Screen. For that, we need to import the `AdMobBanner` component in the Home.js file as directed in the code snippet below:
+
+```json
+import {
+  AdMobBanner,
+  setTestDeviceIDAsync
+} from "expo-ads-admob";
+```
+
+- `AdMobBanner` The component that delivers the Banner Ad to the interface.
+- `setTestDeviceIDAsync`This module enables us to display ads on test environment.
+
+First, we are going to call the `setTestDeviceIDAsync` method to a function `initAdmob`. Then, we need to call the `initAdmob` function inside the `useEffect` hook so that it triggers every time the app loads. The coding implementation is provided in the code snippet below:
+
+```jsx
+const Home = () => {
+  React.useEffect(() => {
+    await setTestDeviceIDAsync("EMULATOR");
```

Comment:
Remove the `await` keyword here. It will throw an error if you use `await` outside of an async function even though the function you are calling is async.

Reply:
oh yep, updated as your mention.